### ISSUE TITLE
API for getting the current text

### ISSUE THREAD

**Issue by boyeln** [RadioRevolt/DABelFish#54]
An API where you can get the current DAB-text as a json object should be developed.

This API should in the future be developed to yield both the next and previous song. This cannot be done before #27.

You should (in the future) also be able to set a new DAB-text through the API, so that an external program (such as liquidsoap) can set the current DAB-text. Only approved external programs should be able to use the API to set a new DAB-text. API-keys might be a solution, but a more secure method should be developed. The DAB-text website (at least the API) should be on https (#55)

**Comment by tobinus** [comment]
Is it really necessary to create a simplified API, or would it suffice to require clients to use [a DDP client](http://www.meteorpedia.com/read/DDP_Clients)? We could create users with read-only access (new role?), and let applications use those. By using a DDP client, the client will always be updated and we minimize the amount of work (although we increase complexity for the clients).

If a new (REST) API is to be developed, there are a few pointers at [the official documentation](http://guide.meteor.com/data-loading.html#publications-as-rest) as well as [the wiki](http://www.meteorpedia.com/read/REST_API) :cake:

**Comment by boyeln** [comment]
> Is it really necessary to create a simplified API, or would it suffice to require clients to use a DDP client? 

No, that was just the first solution that came to mind. The DDP solution seems to be a much better alternative for database manipulations.

But keep in mind that it would be nice for external things, such as the new website (radiorevolt.no) etc, to have easy access to current and previous DAB-texts.

[Graphql](http://graphql.org/) might be a more elegant and simpler solution than a standard REST(ish) API.
